0.34.0: Some tests use the internet and are therefore flaky

### Issue description

Hi! 👋 

When upgrading the package to 0.34.0 on Arch Linux I ran into some issues with a few tests.

```
=========================== short test summary info ============================
FAILED test/widgets/test_generic_poll_text.py::test_gen_poll_url_text - AssertionError: assert 'anything' in '<html>\r\n<head><title>503 Service Temporarily Unavailable</title></head>\r\n<bod
y>\r\n<center><h1>503 Service Temporarily Unavailable</h1></center>\r\n</body>\r\n</html>\r\n'
FAILED test/widgets/test_generic_poll_text.py::test_gen_poll_url_json_with_data - assert "Can't parse" == '{"test": "value"}'

  - {"test": "value"}
  + Can't parse
FAILED test/widgets/test_generic_poll_text.py::test_gen_poll_url_custom_headers - assert 'X-Custom-Header' in "Can't parse"
```


[qtile-0.34.0-1-x86_64-build.log](https://github.com/user-attachments/files/24054136/qtile-0.34.0-1-x86_64-build.log)
[qtile-0.34.0-1-x86_64-check.log](https://github.com/user-attachments/files/24054135/qtile-0.34.0-1-x86_64-check.log)

I guess if they are not run locally, they are bound to be flaky (I will disable these tests).
In general it would be great to not require access to the internet during tests.

### Version

0.34.0

### Backend

X11 (default)

### Config

```python
n/a
```

### Logs

```text
see above
```

### Required

- [x] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [x] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -20,10 +20,6 @@ jobs:
                 backend: ['x11', 'wayland']
         steps:
             - uses: actions/checkout@v4
-            - name: Start httpbin container
-              run: |
-                docker run -d --name httpbin -p 8080:80 kennethreitz/httpbin
-                timeout 30 bash -c 'until curl -s http://localhost:8080/get > /dev/null; do sleep 1; done'
             - name: Set up python ${{ matrix.python-version }}
               uses: actions/setup-python@v5
               with:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -47,6 +47,7 @@ dev = [
     "coverage",
     "pytest-cov",
     "pytest-asyncio>=1.1.0",
+    "pytest-httpbin",
     "pre-commit",
     "PyGObject",
     "pycairo >= 1.25.1",

--- a/test/widgets/test_generic_poll_text.py
+++ b/test/widgets/test_generic_poll_text.py
@@ -1,16 +1,7 @@
-import os
-
 import pytest
 
 import libqtile
 from libqtile.widget import generic_poll_text
-
-# Use local httpbin in CI, fallback to httpbin.org for local development
-HTTPBIN_BASE = (
-    "http://localhost:8080"
-    if os.environ.get("GITHUB_ACTIONS") == "true"
-    else "https://httpbin.org"
-)
 
 
 def test_gen_poll_text():
@@ -46,9 +37,9 @@ def test_gen_poll_url_headers_and_json():
 
 
 @pytest.mark.asyncio
-async def test_gen_poll_url_text():
+async def test_gen_poll_url_text(httpbin):
     gpurl = generic_poll_text.GenPollUrl(
-        json=False, parse=lambda x: x, url=f"{HTTPBIN_BASE}/anything"
+        json=False, parse=lambda x: x, url=f"{httpbin.url}/anything"
     )
     result = await gpurl.apoll()
     assert isinstance(result, str)
@@ -56,38 +47,38 @@ async def test_gen_poll_url_text():
 
 
 @pytest.mark.asyncio
-async def test_gen_poll_url_json_with_data():
+async def test_gen_poll_url_json_with_data(httpbin):
     gpurl = generic_poll_text.GenPollUrl(
-        parse=lambda x: x["data"], data={"test": "value"}, url=f"{HTTPBIN_BASE}/anything"
+        parse=lambda x: x["data"], data={"test": "value"}, url=f"{httpbin.url}/anything"
     )
     result = await gpurl.apoll()
     assert result == '{"test": "value"}'
 
 
 @pytest.mark.asyncio
-async def test_gen_poll_url_xml_no_xmltodict():
+async def test_gen_poll_url_xml_no_xmltodict(httpbin):
     gpurl = generic_poll_text.GenPollUrl(
-        json=False, xml=True, parse=lambda x: x, url=f"{HTTPBIN_BASE}/anything"
+        json=False, xml=True, parse=lambda x: x, url=f"{httpbin.url}/anything"
     )
     result = await gpurl.apoll()
     assert result == "Can't parse"
 
 
 @pytest.mark.asyncio
-async def test_gen_poll_url_broken_parse():
+async def test_gen_poll_url_broken_parse(httpbin):
     gpurl = generic_poll_text.GenPollUrl(
-        json=False, parse=lambda x: x.foo, url=f"{HTTPBIN_BASE}/anything"
+        json=False, parse=lambda x: x.foo, url=f"{httpbin.url}/anything"
     )
     result = await gpurl.apoll()
     assert result == "Can't parse"
 
 
 @pytest.mark.asyncio
-async def test_gen_poll_url_custom_headers():
+async def test_gen_poll_url_custom_headers(httpbin):
     gpurl = generic_poll_text.GenPollUrl(
         headers={"X-Custom-Header": "test-value", "X-Another-Header": "another-value"},
         parse=lambda x: x["headers"],
-        url=f"{HTTPBIN_BASE}/headers",
+        url=f"{httpbin.url}/headers",
     )
     result = await gpurl.apoll()
     assert "X-Custom-Header" in result

